base/linalg/triangular.jl

```diff
@@ -376,6 +376,8 @@ scale!(c::Number, A::Union{UpperTriangular,LowerTriangular}) = scale!(A,c)
 ######################
 
 A_mul_B!(A::Tridiagonal, B::AbstractTriangular) = A*full!(B)
+A_mul_B!(C::AbstractMatrix, A::AbstractTriangular, B::Tridiagonal) = A_mul_B!(C, full(A), B)
```

Comment:
I'm not sure that it's a good idea to have all these `!` methods calling `full`. It used to be the case that (with very few exceptions) `A_mul_B!` didn't allocate for dense matrices and that these methods were the low-level "expert" methods. Therefore, it might be better to have these conversion at a higher level, i.e. `*`, and `Ax_mul_B`.